lib/gollum-lib/file.rb

```diff
@@ -26,6 +26,13 @@ def url_path
       path
     end
 
+    # Public: The SHA hash identifying this file
+    #
+    # Returns the String SHA.
+    def sha
+      @blob && @blob.id
```

Comment:
Is `@blob` ever nil for a Gollum::File? I suppose they might be if we start serving uncommitted files, right? Otherwise the `@blob` check is superfluous.